### ISSUE TITLE
incorrect "warning: unused import" compiling rayon

### ISSUE THREAD

**Issue by cuviper** [rust-lang/rust#38104]
I'm getting incorrect warnings compiling Rayon from git (commit 5b27a6c29b01).  Many like:

```
warning: unused import: `super::internal::*;`, #[warn(unused_imports)] on by default
 --> src/par_iter/option.rs:2:5
  |
2 | use super::internal::*;
  |     ^^^^^^^^^^^^^^^^^^^
```

I only get this using nightly, `rustc 1.15.0-nightly (1c448574b 2016-11-28)`, and it's happy if I remove that import.  However, neither stable nor beta have a warning here, and with that removed they get hard errors about unresolved names and traits not in scope.

I do expect we need the imports, so nightly's behavior is surprising.

**Comment by bluss** [comment]
I'm going to guess that `use super::*;` (the line before in [the file](https://github.com/nikomatsakis/rayon/blob/master/src/par_iter/option.rs)) is now bringing in all symbols the file needs, because globs have changed with now stable `feature(item_like_imports)`.

**Comment by cuviper** [comment]
Oh... the `super` *also* has a [`use internal::*`](https://github.com/nikomatsakis/rayon/blob/e2d8593dd0d870d54b1c7cd92342560f830f035d/src/par_iter/mod.rs#L28) -- so now that adds to `super::*`?

OK, I guess this is NOTABUG, but we'll have to figure out how to restructure those imports to work cleanly with both stable and nightly rustc.
